src/emulate.c

```diff
@@ -19,11 +20,14 @@
 #include "softfloat.h"
 #endif /* RV32_HAS(EXT_F) */
 
+static void intr_excp_handle(riscv_t *rv);
```

Comment:
The name looks weird. Can you unify as exception handler?

Reply:
For minimal changes, the mixture of interrupt and exception in a single structure is really confusing. RISC-V does have its exception definitions as well as the counterpart for interrupts.

Reply:
By default, M-mode is the first receiver of any interrupts or exceptions (i.e., traps) in the system. Instead of weird "intr_excp_handle," we can call it "trap_handler." The latter is used in semu.

Reply:
> For minimal changes, the mixture of interrupt and exception in a single structure is really confusing. RISC-V does have its exception definitions as well as the counterpart for interrupts.

Yes, we can determine a trap is an exception or an interrupt via the MSB of the trap code.